rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,300 @@
+
+V2_SWAP = b'!p\xc7A\xc4\x151\xae\xc2\x0e|\x10|$\xee\xcf\xdd\x15\xe6\x9c\x9b\xb0\xa8\xdd7\xb1\x84\x0b\x9e\x0b {'  # noqa: E501
+JOIN_V1 = b'c\x98-\xf1\x0e\xfd\x8d\xfa\xaa\xa0\xfc\xc7\xf5\x0b-\x93\xb7\xcb\xa2l\xccH\xad\xee(s"\rH]\xc3\x9a'  # noqa: E501
+EXIT_V1 = b'\xe7L\x91U+d\xc2\xe2\xe7\xbd%V9\xe0\x04\xe6\x93\xbd>\x1d\x01\xcc3\xe6V\x10\xb8j\xfc\xc1\xff\xed'  # noqa: E501
+VAULT_ADDRESS = string_to_evm_address('0xBA12222222228d8Ba445958a75a0704d566BF2C8')
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BalancerDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.base_tools = base_tools
+        self.eth = A_ETH.resolve_to_crypto_asset()
+        self.weth = A_WETH.resolve_to_evm_token()
+
+    def _decode_v1_pool_event(self, all_logs: list[EvmTxReceiptLog]) -> Optional[list[ActionItem]]:
+        """Read the list of logs in search for a Balancer V1 event and return the information
+        needed to decode the transfers made in the transaction to/from the ds proxy
+        """
+        # The transfer event appears after the debt generation event, so we need to transform it
+        target_logs = []
+        for tx_log in all_logs:
+            if tx_log.topics[0] == JOIN_V1 or tx_log.topics[0] == EXIT_V1:
+                target_logs.append(tx_log)
+
+        if len(target_logs) == 0:
+            return None
+
+        action_items = []
+        for target_log in target_logs:
+            token_address = hex_or_bytes_to_address(target_log.topics[2])
+            amount = hex_or_bytes_to_int(target_log.data[0:32])
+            if target_log.topics[0] == JOIN_V1:
+                balancer_event_type = BalancerV1EventTypes.JOIN
+                from_event_type = HistoryEventType.SPEND
+                ds_address = hex_or_bytes_to_address(target_log.topics[1])
+            else:
+                balancer_event_type = BalancerV1EventTypes.EXIT
+                from_event_type = HistoryEventType.RECEIVE
+                ds_address = target_log.address
+
+            action_item = ActionItem(
+                action='transform',
+                sequence_index=target_log.log_index,
+                from_event_type=from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=self.eth,
+                amount=ZERO,
+                to_event_type=None,
+                to_event_subtype=None,
+                to_counterparty=CPT_BALANCER_V1,
+                to_notes=None,
+                extra_data={
+                    'ds_address': ds_address,
+                    'token_address': token_address,
```

Comment:
Why not keep it at the `asset` of the `ActionItem`? Do you specifically need an address or an asset?

Reply:
you have not answered this

Reply:
I have changed this function to return a dictionary instead. I was using actions before because of how I was working with the events but saw that it could be done easier just calling the logic in the `_decode_v1_pool_event` function. It doesn't need and it shouldn't be an action